custom/enikshay/management/commands/nikshay_registration_notification_time_report.py

```diff
@@ -0,0 +1,73 @@
+        self.since = (datetime.datetime(self.till.year, self.till.month, self.till.day)
+                      - datetime.timedelta(days=days)
+                      - datetime.timedelta(hours=5, minutes=30))
+
+        result_file_name = "nikshay_registration_notification_time_report_from_%s_till_%s.csv" % (
+            self.since.strftime('%Y-%m-%d-%H:%M:%S'),
+            self.till.strftime('%Y-%m-%d-%H:%M:%S')
+        )
+
+        with open(result_file_name, 'w') as csvfile:
+            writer = csv.DictWriter(csvfile,
+                                    fieldnames=["nikshay id", "form finished on", "form submitted on",
+                                                "notification completed on", "form to submission",
+                                                "submission to notification", "enikshay id"])
+            writer.writeheader()
+            case_accessor = CaseAccessors(DOMAIN)
+            for repeat_record in iter_repeat_records_by_domain(DOMAIN, repeater_id=REGISTRATION_REPEATER_ID,
+                                                               state=SUCCESS_STATE, since=self.since):
+                episode_case_id = repeat_record.payload_id
+                episode_case = case_accessor.get_case(episode_case_id)
+                assert repeat_record.succeeded
+                time_of_notification = pytz.utc.localize(repeat_record.last_checked).astimezone(timezone)
+                # assert that
+                # the last notification was the success one and
+                # the time for last notification is same as that for the repeat record
+                last_notification_attempt = repeat_record.attempts[-1]
+                assert last_notification_attempt.succeeded
+                assert repeat_record.last_checked == last_notification_attempt.datetime
+                property_changed_info = get_case_property_changed_info(episode_case,
+                                                                       "treatment_initiated",
+                                                                       "yes_phi")
+                xform = property_changed_info.case_update.form
+                form_received_on = pytz.utc.localize(xform.received_on).astimezone(timezone)
+                property_modified_on = parse_datetime(property_changed_info.modified_on).astimezone(timezone)
+                writer.writerow({
+                    'nikshay id': episode_case.get_case_property('nikshay_id'),
+                    'form finished on': property_modified_on.strftime('%Y-%m-%d-%H:%M:%S'),
+                    'form submitted on': form_received_on.strftime('%Y-%m-%d-%H:%M:%S'),
+                    'notification completed on': time_of_notification.strftime('%Y-%m-%d-%H:%M:%S'),
+                    'form to submission': (form_received_on - property_modified_on),
+                    'submission to notification': (time_of_notification - form_received_on),
+                    'enikshay id': episode_case.case_id
```

Comment:
This isn't the enikshay_id. Normally, they would want the "readable ID" or the case_id of the person case.

Reply:
This is for reference to quickly get to the case detail page. Renaming it to case id to avoid confusion.